build.sbt

```diff
@@ -1,42 +1,52 @@
 import sbt._
-import org.allenai.plugins.CoreDependencies
-import org.allenai.plugins.CoreDependencies._
-import sbtrelease._
 import sbtrelease.ReleaseStateTransformations._
 
 name := "Word2VecJava"
 
-libraryDependencies ++= Seq(
-  "org.apache.commons" % "commons-lang3" % "3.1",
-  "com.google.guava" % "guava" % "18.0",
-  "commons-io" % "commons-io" % "2.4",
-  "log4j" % "log4j" % "1.2.17",
-  "joda-time" % "joda-time" % "2.3",
-  "org.apache.thrift" % "libfb303" % "0.9.1",
-  "org.apache.commons" % "commons-math3" % "3.4.1",
-  "org.scalacheck" %% "scalacheck" % "1.12.0" % Test,
-  "com.novocode" % "junit-interface" % "0.11" % Test)
+lazy val scala212 = "2.12.9"
+lazy val scala211 = "2.11.12"
+lazy val scala213 = "2.13.0"
+lazy val supportedScalaVersions = List(scala212, scala211, scala213)
 
-// Override the problematic new release plugin.
-lazy val releaseProcessSetting = releaseProcess := Seq(
-  checkSnapshotDependencies,
-  inquireVersions,
-  runClean,
-  runTest,
-  setReleaseVersion,
-  commitReleaseVersion,
-  tagRelease,
-  publishArtifacts,
-  setNextVersion,
-  commitNextVersion,
-  pushChanges
-)
+ThisBuild / organization := "org.allenai.word2vec"
+ThisBuild / scalaVersion := scala212
 
+lazy val root = (project in file("."))
+    .settings(
+      crossScalaVersions := supportedScalaVersions,
+      name := "Word2VecJava",
+      buildSettings,
+      libraryDependencies ++= Seq(
+        "org.apache.commons" % "commons-lang3" % "3.9",
+        "com.google.guava" % "guava" % "18.0",
+        "commons-io" % "commons-io" % "2.4",
+        "log4j" % "log4j" % "1.2.17",
+        "joda-time" % "joda-time" % "2.3",
+        "org.apache.thrift" % "libfb303" % "0.9.3",
+        "org.apache.commons" % "commons-math3" % "3.6.1",
+        "org.scalacheck" %% "scalacheck" % "1.14.0" % Test,
+        "com.novocode" % "junit-interface" % "0.11" % Test
+      ),
+
+      // don't use sbt-release's cross facility
+      releaseCrossBuild := false,
```

Comment:
I am ever an sbt-neophyte -- could you give me a high level explanation of how this works? I recognize you want to produce an artifact(s?) compatible with multiple versions of Scala, but I'm unclear on how we arrive there with this. How does turning off `releaseCrossBuild` reach the goal?

Reply:
This line specifically is a hack to fix an incompatibility between SBT 1.0's "+" functionality and the `sbt-release` plugin's "+" functionality which is ported from 0.13. You can read more about it on the sbt docs:

https://www.scala-sbt.org/1.x/docs/Cross-Build.html#Note+about+sbt-release

The below complete list of steps include "+test" which is like the "test" target except it runs for each of the configured cross-compilation Scala versions. In this case that's 3 versions.

Similarly for +publishSigned.